Invalid Datatype: Expected NodeID , Recevied ExpandedNodeID

I am building an OPC UA server. I can't set the data type of the node as shown in the UA Expert image below.

<img width="632" height="300" alt="Image" src="https://github.com/user-attachments/assets/9afcb2d5-ced2-4c4e-a627-535ba2a52043" />

Here is the code I am using

```go
func (s *Server) createVariableNode(dbNode *models.Node) *server.Node {
	nodeID := ua.NewStringNodeID(uint16(s.namespace.ID()), dbNode.BrowseName)

	accessLevel := byte(ua.AccessLevelTypeCurrentRead)
	if dbNode.IsWritable {
		accessLevel = byte(ua.AccessLevelTypeCurrentRead | ua.AccessLevelTypeCurrentWrite)
	}

	valueGetter := s.createValueGetter(dbNode)

	attributes := map[ua.AttributeID]*ua.DataValue{
		ua.AttributeIDAccessLevel:     server.DataValueFromValue(accessLevel),
		ua.AttributeIDUserAccessLevel: server.DataValueFromValue(accessLevel),
		ua.AttributeIDBrowseName:      server.DataValueFromValue(attrs.BrowseName(dbNode.BrowseName)),
		ua.AttributeIDNodeClass:       server.DataValueFromValue(uint32(ua.NodeClassVariable)),
		ua.AttributeIDValueRank:       server.DataValueFromValue(int32(-1)), // Scalar
	}

	n := server.NewNode(nodeID, attributes, nil, valueGetter)

	n.SetAttribute(ua.AttributeIDDataType, server.DataValueFromValue(&ua.ExpandedNodeID{
		NodeID: ua.NewNumericNodeID(0, id.Double),
	}))

	return n
}
```

console error
```bash
panic: interface conversion: interface {} is *ua.NodeID, not *ua.ExpandedNodeID

goroutine 1 [running]:
github.com/gopcua/opcua/server.(*Node).DataType(0xc002cc2740)
        C:/Users/_/go/pkg/mod/github.com/gopcua/opcua@v0.8.0/server/node.go:280 +0x21c
github.com/gopcua/opcua/server.(*Node).AddRef(0xc000f6b180, 0xc002cc2740, 0x2f, 0x1)
        C:/Users/_/go/pkg/mod/github.com/gopcua/opcua@v0.8.0/server/node.go:351 +0x212
opcua-server/internal/opcua.(*Server).loadNodesFromDB(0xc0018802a0, {0xd94a10?, 0xc0004ba2d0?}, 0xc000f6b180)
        D:/Work/opcua-server-go/internal/opcua/server.go:200 +0x79f
opcua-server/internal/opcua.(*Server).Start(0xc0018802a0, {0xd94a10, 0xc0004ba2d0})
        D:/Work/opcua-server-go/internal/opcua/server.go:149 +0x1e7
main.main()
        D:/Work/opcua-server-go/cmd/server/main.go:109 +0x8f9
exit status 2
make: *** [Makefile:4: run] Error 1
```

--- a/server/namespace_node.go
+++ b/server/namespace_node.go
@@ -130,32 +130,31 @@ func (as *NodeNameSpace) Attribute(id *ua.NodeID, attr ua.AttributeID) *ua.DataV
 		return errorDataValueWithStatus(ua.StatusBadUserAccessDenied)
 	}
 
-	var err error
-	var a *AttrValue
-
 	switch attr {
 	case ua.AttributeIDNodeID:
-		a = &AttrValue{Value: DataValueFromValue(id)}
+		return DataValueFromValue(id)
 	case ua.AttributeIDEventNotifier:
 		// TODO: this is a hack to force the EventNotifier to false for everything.
 		// If at some point someone or something needs to use this, this will have to go away and be
 		// fixed properly.
-		a = &AttrValue{Value: DataValueFromValue(byte(0))}
-	case ua.AttributeIDNodeClass:
-		if a, err = n.Attribute(attr); err != nil {
-			return errorDataValueWithStatus(ua.StatusBadAttributeIDInvalid)
-		}
+		return DataValueFromValue(byte(0))
+	}
 
+	var err error
+	var a *AttrValue
+
+	if a, err = n.Attribute(attr); err != nil {
+		return errorDataValueWithStatus(ua.StatusBadAttributeIDInvalid)
+	}
+
+	switch attr {
+	case ua.AttributeIDNodeClass:
 		// TODO: we need int32 instead of uint32 here.  this isn't the right place to fix it, but it is a bandaid
 		x, ok := a.Value.Value.Value().(uint32)
 		if ok {
 			a.Value.Value = ua.MustVariant(int32(x))
 		}
 	case ua.AttributeIDDataType:
-		if a, err = n.Attribute(attr); err != nil {
-			return errorDataValueWithStatus(ua.StatusBadAttributeIDInvalid)
-		}
-
 		if a.Value != nil && a.Value.Value != nil {
 			if nodeID := a.Value.Value.NodeID(); nodeID != nil {
 				dv := *a.Value
@@ -165,10 +164,6 @@ func (as *NodeNameSpace) Attribute(id *ua.NodeID, attr ua.AttributeID) *ua.DataV
 		}
 
 		return errorDataValueWithStatus(ua.StatusBadTypeMismatch)
-	default:
-		if a, err = n.Attribute(attr); err != nil {
-			return errorDataValueWithStatus(ua.StatusBadAttributeIDInvalid)
-		}
 	}
 
 	return a.Value

--- a/server/namespace_node.go
+++ b/server/namespace_node.go
@@ -133,6 +133,8 @@ func (as *NodeNameSpace) Attribute(id *ua.NodeID, attr ua.AttributeID) *ua.DataV
 	var err error
 	var a *AttrValue
 
+	status := ua.StatusOK
+
 	switch attr {
 	case ua.AttributeIDNodeID:
 		a = &AttrValue{Value: DataValueFromValue(id)}
@@ -143,29 +145,42 @@ func (as *NodeNameSpace) Attribute(id *ua.NodeID, attr ua.AttributeID) *ua.DataV
 		a = &AttrValue{Value: DataValueFromValue(byte(0))}
 	case ua.AttributeIDNodeClass:
 		a, err = n.Attribute(attr)
-		if err != nil {
-			return &ua.DataValue{
-				EncodingMask:    ua.DataValueServerTimestamp | ua.DataValueStatusCode,
-				ServerTimestamp: time.Now(),
-				Status:          ua.StatusBadAttributeIDInvalid,
+		if err == nil {
+			// TODO: we need int32 instead of uint32 here.  this isn't the right place to fix it, but it is a bandaid
+			x, ok := a.Value.Value.Value().(uint32)
+			if ok {
+				a.Value.Value = ua.MustVariant(int32(x))
 			}
 		}
-		// TODO: we need int32 instead of uint32 here.  this isn't the right place to fix it, but it is a bandaid
-		x, ok := a.Value.Value.Value().(uint32)
-		if ok {
-			a.Value.Value = ua.MustVariant(int32(x))
+	case ua.AttributeIDDataType:
+		if a, err = n.Attribute(attr); err != nil {
+			status = ua.StatusBadAttributeIDInvalid
+			break
 		}
+
+		if a.Value != nil && a.Value.Value != nil {
+			if nodeID := a.Value.Value.NodeID(); nodeID != nil {
+				dv := *a.Value
+				dv.Value = ua.MustVariant(nodeID)
+				return &dv
+			}
+		}
+
+		status = ua.StatusBadTypeMismatch
 	default:
-		a, err = n.Attribute(attr)
+		if a, err = n.Attribute(attr); err != nil {
+			status = ua.StatusBadAttributeIDInvalid
+		}
 	}
 
-	if err != nil {
+	if status != ua.StatusOK {
 		return &ua.DataValue{
 			EncodingMask:    ua.DataValueServerTimestamp | ua.DataValueStatusCode,
 			ServerTimestamp: time.Now(),
-			Status:          ua.StatusBadAttributeIDInvalid,
+			Status:          status,
 		}
 	}
+
 	return a.Value
 }
 

--- a/server/namespace_node.go
+++ b/server/namespace_node.go
@@ -113,27 +113,25 @@ func (as *NodeNameSpace) AddNewVariableStringNode(name string, value any) *Node 
 }
 
 func (as *NodeNameSpace) Attribute(id *ua.NodeID, attr ua.AttributeID) *ua.DataValue {
-	n := as.Node(id)
-	if n == nil {
+	errorDataValueWithStatus := func(status ua.StatusCode) *ua.DataValue {
 		return &ua.DataValue{
 			EncodingMask:    ua.DataValueServerTimestamp | ua.DataValueStatusCode,
 			ServerTimestamp: time.Now(),
-			Status:          ua.StatusBadNodeIDUnknown,
+			Status:          status,
 		}
 	}
 
+	n := as.Node(id)
+	if n == nil {
+		return errorDataValueWithStatus(ua.StatusBadNodeIDUnknown)
+	}
+
 	if !n.Access(ua.AccessLevelTypeCurrentRead) {
-		return &ua.DataValue{
-			EncodingMask:    ua.DataValueServerTimestamp | ua.DataValueStatusCode,
-			ServerTimestamp: time.Now(),
-			Status:          ua.StatusBadUserAccessDenied,
-		}
+		return errorDataValueWithStatus(ua.StatusBadUserAccessDenied)
 	}
 
 	var err error
 	var a *AttrValue
-
-	status := ua.StatusOK
 
 	switch attr {
 	case ua.AttributeIDNodeID:
@@ -144,18 +142,18 @@ func (as *NodeNameSpace) Attribute(id *ua.NodeID, attr ua.AttributeID) *ua.DataV
 		// fixed properly.
 		a = &AttrValue{Value: DataValueFromValue(byte(0))}
 	case ua.AttributeIDNodeClass:
-		a, err = n.Attribute(attr)
-		if err == nil {
-			// TODO: we need int32 instead of uint32 here.  this isn't the right place to fix it, but it is a bandaid
-			x, ok := a.Value.Value.Value().(uint32)
-			if ok {
-				a.Value.Value = ua.MustVariant(int32(x))
-			}
+		if a, err = n.Attribute(attr); err != nil {
+			return errorDataValueWithStatus(ua.StatusBadAttributeIDInvalid)
+		}
+
+		// TODO: we need int32 instead of uint32 here.  this isn't the right place to fix it, but it is a bandaid
+		x, ok := a.Value.Value.Value().(uint32)
+		if ok {
+			a.Value.Value = ua.MustVariant(int32(x))
 		}
 	case ua.AttributeIDDataType:
 		if a, err = n.Attribute(attr); err != nil {
-			status = ua.StatusBadAttributeIDInvalid
-			break
+			return errorDataValueWithStatus(ua.StatusBadAttributeIDInvalid)
 		}
 
 		if a.Value != nil && a.Value.Value != nil {
@@ -166,18 +164,10 @@ func (as *NodeNameSpace) Attribute(id *ua.NodeID, attr ua.AttributeID) *ua.DataV
 			}
 		}
 
-		status = ua.StatusBadTypeMismatch
+		return errorDataValueWithStatus(ua.StatusBadTypeMismatch)
 	default:
 		if a, err = n.Attribute(attr); err != nil {
-			status = ua.StatusBadAttributeIDInvalid
-		}
-	}
-
-	if status != ua.StatusOK {
-		return &ua.DataValue{
-			EncodingMask:    ua.DataValueServerTimestamp | ua.DataValueStatusCode,
-			ServerTimestamp: time.Now(),
-			Status:          status,
+			return errorDataValueWithStatus(ua.StatusBadAttributeIDInvalid)
 		}
 	}
 

--- a/server/node_test.go
+++ b/server/node_test.go
@@ -37,3 +37,25 @@ func TestNamespaceAttributeDataTypeReturnsNodeID(t *testing.T) {
 	require.Equal(t, ua.TypeIDNodeID, dv.Value.Type())
 	require.Equal(t, uint32(id.Double), dv.Value.NodeID().IntID())
 }
+
+func TestNamespaceAttributeMissingNodeClassReturnsBadAttributeIDInvalid(t *testing.T) {
+	ns := NewNameSpace("test")
+	nodeID := ua.NewStringNodeID(1, "x")
+
+	n := NewNode(
+		nodeID,
+		Attributes{
+			ua.AttributeIDBrowseName:  DataValueFromValue(&ua.QualifiedName{Name: "x"}),
+			ua.AttributeIDDisplayName: DataValueFromValue(&ua.LocalizedText{Text: "x"}),
+		},
+		nil,
+		nil,
+	)
+
+	ns.AddNode(n)
+
+	require.NotPanics(t, func() {
+		dv := ns.Attribute(nodeID, ua.AttributeIDNodeClass)
+		require.Equal(t, ua.StatusBadAttributeIDInvalid, dv.Status)
+	})
+}

--- a/server/node_test.go
+++ b/server/node_test.go
@@ -1,0 +1,39 @@
+// Copyright 2018-2020 opcua authors. All rights reserved.
+// Use of this source code is governed by a MIT-style license that can be
+// found in the LICENSE file.
+
+package server
+
+import (
+	"testing"
+
+	"github.com/gopcua/opcua/id"
+	"github.com/gopcua/opcua/ua"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNamespaceAttributeDataTypeReturnsNodeID(t *testing.T) {
+	ns := NewNameSpace("test")
+	nodeID := ua.NewStringNodeID(1, "x")
+
+	n := NewNode(
+		nodeID,
+		Attributes{
+			ua.AttributeIDBrowseName:  DataValueFromValue(&ua.QualifiedName{Name: "x"}),
+			ua.AttributeIDDisplayName: DataValueFromValue(&ua.LocalizedText{Text: "x"}),
+			ua.AttributeIDNodeClass:   DataValueFromValue(uint32(ua.NodeClassVariable)),
+			ua.AttributeIDDataType:    DataValueFromValue(ua.NewNumericExpandedNodeID(0, id.Double)),
+		},
+		nil,
+		nil,
+	)
+
+	ns.AddNode(n)
+
+	dv := ns.Attribute(nodeID, ua.AttributeIDDataType)
+
+	require.Equal(t, ua.StatusOK, dv.Status)
+	require.NotNil(t, dv.Value.NodeID())
+	require.Equal(t, ua.TypeIDNodeID, dv.Value.Type())
+	require.Equal(t, uint32(id.Double), dv.Value.NodeID().IntID())
+}

